lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2433,8 +2433,48 @@ ParseResult FIRStmtParser::parseInstance() {
   ArrayAttr annotations = getAnnotations(getModuleTarget() + ">" + id);
   auto name = hasDontTouch(annotations) ? id : filterUselessName(id);
 
-  auto result =
-      builder.create<InstanceOp>(resultTypes, moduleName, name, annotations);
+  auto result = builder.create<InstanceOp>(resultTypes, moduleName, name);
+
+  // Annotations targeting results of instance ops should be attached through
+  // the setPortAnnotation hook.
+  SmallVector<Attribute, 16> AnnotationsVec;
+  for (auto attr : annotations) {
+    auto dictAttr = attr.cast<DictionaryAttr>();
+    auto targetAttr = dictAttr.get("target");
+    if (!targetAttr) {
+      AnnotationsVec.push_back(attr);
+      continue;
+    }
+
+    // The first token of targetAttr should always begin with '.'.
+    auto targetAttrValue = targetAttr.cast<ArrayAttr>().getValue();
+    auto portName = targetAttrValue[0].cast<StringAttr>().getValue();
+    if (portName.front() != '.') {
+      return emitError(startTok.getLoc(),
+                       "unexpected annotation target " + portName);
+    }
+
+    // If there's more than one tokens, the remaining tokens should be stored as
+    // target in the result annotation.
+    SmallVector<NamedAttribute, 8> dictAttrVec;
+    for (auto namedAttr : dictAttr) {
+      if (namedAttr.first == "target") {
+        if (targetAttrValue.size() > 1) {
+          auto newTargetAttr =
+              builder.getArrayAttr(targetAttrValue.drop_front());
+          dictAttrVec.push_back(builder.getNamedAttr("target", newTargetAttr));
+        }
+      } else
+        dictAttrVec.push_back(namedAttr);
+    }
+
+    if (!result.setPortAnnotation(portName.drop_front(),
+                                  builder.getDictionaryAttr(dictAttrVec)))
```

Comment:
Please set the port annotations once at the end instead of repeatedly calling setPortAnnotation (and also, please work by index, not by name).  The difference here is that each intermediate "set" will create a new ArrayAttr on the outside level and MLIR uniques these, so they will never get deallocated.